Training24/Test.c

```diff
@@ -1,14 +1,193 @@
 // ------------------------------------------------------------------------------------------------
 // Training ~ A training program for new joiners at Metamation, Batch - July 2024.
 // Copyright (c) Metamation India.
+// Kalaimagal V P
 // ------------------------------------------------------------------------------------------------
 // Test.c
-// Program on main branch.
+// Program on A4 branch.
+// 1. Palindrome is a word that reads the same forward or backward. Example: Civic, Madam, Bib, Nun, Level and so on. 
+//    Your program must take the word as input and display the result if it is a palindrome or not. 
+//    For complexity, you can also use phrases instead of words (E.g., Was it a car or a cat I saw? I did, did I? Don�t nod)
+// 2. Write a program that reverses a given integer (as defined in C) as explained below and check if it is a palindrome.
 // ------------------------------------------------------------------------------------------------
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <malloc.h>
+#include <ctype.h> 
+#include <stdbool.h>
+#include <conio.h> // For getch()
 #include "Header.h"
 
-int main () {
-   printf ("%s\n", GetString ());
-   return 0;
+// ANSI escape codes for colors
+#define RESET   "\033[0m"
+#define GREEN   "\033[32m"
+#define RED     "\033[31m"
+#define YELLOW  "\033[33m"
+#define MAGENTA "\033[35m"
+#define SKYBLUE "\033[36m"
+
+/// <summary>Function to clear the console screen.</summary>
+void ClearScreen () {
+#ifdef _WIN32
+   system ("cls");           // Windows-specific command to clear the screen
+#endif
+}
+
+void FilterChar (char* buffer, const char* temp) {
+   int j = 0;
+   for (int i = 0; temp[i] != '\0'; i++) if (isalnum (temp[i])) buffer[j++] = tolower (temp[i]); // Convert to lowercase and add to buffer
+   buffer[j] = '\0';
+}
+
+/// <summary>Function to print test results for string.</summary>
+void FormatResult (int numTests, const char** inputs, const char** actualResults, const int* pass) {

```

Comment:
If you have intellisense showing in your code, you can try to see if the suggestion fits and incorporate it accordingly.
![image](https://github.com/user-attachments/assets/40dd4c1e-1882-4c2d-912b-79024ce0e129)
```suggestion
static void FormatResult (int numTests, const char** inputs, const char** actualResults, const int* pass) {

```

Reply:
Hover over the 3 dots to see suggested changes.